### DISCUSSION TITLE
Current status of Photon Monte Carlo

### DISCUSSION THREAD

**Discussion by edwardwang1** [e0404/matRad#774] [General]
Hello,

My research is focused on lung RT, and I think I have reached the limitations of the pencil beam kernel so I would like to explore Monte Carlo options. I've been in touch with Eric Christiansen who said that for the lung portion of his project, he used MC via VMC++. However, according to https://github.com/e0404/matRad/issues/351, VMC++ is no longer supported and that ompMC should be used instead. The limitation of ompMC seems to be that one can not use custom machines via phase space files, and also it is not being actively maintained anymore https://github.com/e0404/matRad/issues/598.

It seems that now TOPAS is the standard MC photon engine, is this correct? I was able to run [Example 16](https://github.com/e0404/matRad/blob/dev/examples/matRad_example16_photonMC_MLC.m) in the dev branch. I had to download the required VarianClinaciX_6MV_20x20_aboveMLC_w2.header and VarianClinaciX_6MV_20x20_aboveMLC_w2.phsp from the IAEA website (https://www-nds.iaea.org/phsp/photon/Varian_Clinac_iX_6MV/ https://www-nds.iaea.org/phsp/photon1/).

At this point I have a few questions:
1. I have been in contact with the National Research Council of Canada and they are willing to give me a research license to use VMC++. Would I be able to set up VMC++ with my custom machine?
2. One of the physics associates at my centre has created a TrueBeam model in EGSnrc based on phase data supplied by Varian. Currently the plan is to convert the necessary files into something that TOPAS can use, but I wonder whether it will be a lot of work to adapt the matRad-TOPAS pipeline to generate the necessary geometry files for/interface with EGSnrc instead. 

Thanks,
Edward

**Comment by wahln** [comment]
Ragarding 1., we still have the files for the (old) interface. Since we don't have access to VMC++ anymore, we could't maintain it, but with a few updates it should be possible to reactivate.
Regarding 2., introducing a new MC engine is a bit of work but doable. It's mainly a script to write the correct input files. @remocristoforetti currently has some experience on that matter besides me, since he is currently working with FRED for protons. I would not start from the TOPAS engine, since it's a bit cluttered due to the encoded flexibility in choosing modalities, scorers, and so on.

**Reply by edwardwang1** [reply]
Thanks Niklas!

I think I'll try and pursue option 1 for now. I believe that in his thesis, Eric Christiansen modified the original VMC++ implementation in matRad to use custom EGSnrc phase space files. Did those changes ever get integrated into matRad (and therefore you would have it)?

**Reply by edwardwang1** [reply]
Ah I see the code is still in the master branch. I will give it a whirl

**Reply by wahln** [reply]
Modifications by him should be on the dev_VMAT_merge branch, which we will tackle after the coming release. Maybe this might also be the time to bring the VMC++ interface into the new forrmat, but we can only do this with the help of someone having access to VMC++.
